### PR TITLE
Py3: use unittest.mock instead of mock

### PR DESCRIPTION
**Pull request by singingwolfboy**

### CODE REVIEW

**Comment by singingwolfboy** [comment]
Could someone please review this pull request? I just rebased it.

File: tests/test_packages/test_httpretty.py [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
-import mock
+
```
Comment (bblommers) [inline tests/test_packages/test_httpretty.py:4]: This import is unused, as far as I can tell

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @singingwolfboy, thanks for this PR!
Would you mind moving the try-catches to `compat.py`? The file is already in place in both the moto and tests directory - having all compatibility imports in one place will make it easier to keep track of them.

Looks like the CI is failing at the moment btw, but I haven't looked into why. Let me know if you need a hand with that.

**Review by bblommers — CHANGES_REQUESTED** [review]
Nice @singingwolfboy - happy to merge once the build passes

**Comment by singingwolfboy** [comment]
@bblommers I finally got the tests to pass! Can you check my code once again, to make sure that the changes I made are acceptable?

**Comment by bblommers** [comment]
Thanks for getting the build fixed,  @singingwolfboy. 
The try-except brings back some painful memories of #3541, which (I think) fixed the same problem in PY2, introduced by `mock==4.0.3`. 

I'll merge this for now, and have a look later on to see if my patch can be reverted with this solution.

**Review by bblommers — APPROVED** [review]

**Pull request merged**